### ISSUE TITLE
Errors in Public Distributed Api for all_to_all and gather_there

### ISSUE THREAD

**Issue by AndreyKulikov2022** [STEllAR-GROUP/hpx#6544]
## Actual Behavior

In [the actual signature](https://github.com/STEllAR-GROUP/hpx/blob/master/libs/full/collectives/include/hpx/collectives/gather.hpp#L355) of `hpx::collectives::gather_there`:
```c
template <typename T>
    hpx::future<void> gather_there(communicator fid, T&& local_result,
        this_site_arg this_site = this_site_arg(),
        generation_arg generation = generation_arg())
```

the return type differs from the one described in [Public Distributed Api](https://hpx-docs.stellar-group.org/latest/html/libs/full/collectives/api/gather.html#_CPPv4I0EN3hpx11collectives12gather_thereEN3hpx6futureINSt6vectorI7decay_tI1TEEEEEPKcRR1T13this_site_arg14generation_arg13root_site_arg):

```c
 template <typename T>
    hpx::future<std::vector<decay_t<T>>>
    gather_there(communicator comm, T&& result,
        this_site_arg this_site = this_site_arg(),
        generation_arg generation = generation_arg());
```

The same is true for [`hpx::collectives::all_to_all`](https://github.com/STEllAR-GROUP/hpx/blob/master/libs/full/collectives/include/hpx/collectives/all_to_all.hpp#L183):

```c
template <typename T>
    hpx::future<std::vector<std::decay_t<T>>>
    all_to_all(char const* basename, T&& result,
        num_sites_arg num_sites = num_sites_arg(),
        this_site_arg this_site = this_site_arg(),
        generation_arg generation = generation_arg(),
        root_site_arg root_site = root_site_arg());
```

with the wrong local result parameter in [Public Api](https://hpx-docs.stellar-group.org/latest/html/libs/full/collectives/api/all_to_all.html#_CPPv4I0EN3hpx11collectives10all_to_allEN3hpx6futureINSt6vectorINSt7decay_tI1TEEEEEEPKcRR1T13num_sites_arg13this_site_arg14generation_arg13root_site_arg):

```c
template <typename T>
    hpx::future<std::vector<T>> all_to_all(communicator fid,
        std::vector<T>&& local_result,
        this_site_arg this_site = this_site_arg(),
        generation_arg generation = generation_arg())
```

**Comment by hkaiser** [comment]
@AndreyKulikov2022 thanks for reporting this. @dimitraka could you have a look, please?

**Comment by hkaiser** [comment]
This was fixed by merging #6546
